FWCore/TestProcessor/interface/TestProcessor.h

```diff
@@ -378,6 +386,28 @@ This simulates a problem happening early in the job which causes processing not
       bool beginLumiCalled_ = false;
       bool openOutputFilesCalled_ = false;
     };
+
+    template <typename Traits>
+    void TestProcessor::processTransitionForAllStreams(typename Traits::TransitionInfoType& transitionInfo) {
```

Comment:
Since these functions are only called from `TestProcessor.cc, I'd move their definitions there

(then also the `FinalWaitingTask.h` and `WaitingTaskHolder.h` headers would not have to be exposed to all users of `TestProcessor.h`)

Reply:
Done, better. Thanks.